Added controllers to outsource route logic and implement MVC like architecture

Sayan3sarkar opened the pull request:
Hi Anubhav,

I made your code leaner by outsourcing all the logic under each file of the 'routes' directory to the 'controller' directory and simply passed on the controllers as functions to the route declaration middleware. This imitates a MVC like architecture and would make this code easier to understand and follow. Check the files under 'routes' to get a clearer idea of what I did.

Regards,
Sayan Da :)

Sayan3sarkar committed:
Added Controllers for the routes and converted some syntax to ES6+

Sayan3sarkar committed:
Added controllers to outsource route logic and implement MVC like architecture

xprilion commented:
Hi @Sayan3sarkar, thank you for giving the code a beautiful structure! I am getting the following error on trying to run the demo - 

```
Not Found
404
NotFoundError: Not Found
    at /home/xprilion/projects/demoadmin/app.js:46:8
    at Layer.handle [as handle_request] (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:317:13)
    at /home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:335:12)
    at next (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:275:10)
    at /home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:635:15
    at next (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:260:14)
    at Function.handle (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:174:3)
    at router (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:47:12)
```

Steps to reproduce:

1. `npm start`
2. Go to `localhost:1111`
3. Click on **Login**

Can you please see why this might be happening?

Sayan3sarkar commented:
> Hi @Sayan3sarkar, thank you for giving the code a beautiful structure! I am getting the following error on trying to run the demo -
> 
> ```
> Not Found
> 404
> NotFoundError: Not Found
>     at /home/xprilion/projects/demoadmin/app.js:46:8
>     at Layer.handle [as handle_request] (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/layer.js:95:5)
>     at trim_prefix (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:317:13)
>     at /home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:284:7
>     at Function.process_params (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:335:12)
>     at next (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:275:10)
>     at /home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:635:15
>     at next (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:260:14)
>     at Function.handle (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:174:3)
>     at router (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:47:12)
> ```
> 
> Steps to reproduce:
> 
>     1. `npm start`
> 
>     2. Go to `localhost:1111`
> 
>     3. Click on **Login**
> 
> 
> Can you please see why this might be happening?

Hi @xprilion , I'll see as to why this is happening and will get back to you as soon as possible

Sayan3sarkar committed:
Added missing route for GET request to /login route

Sayan3sarkar committed:
Added missing route for GET request to /login route

Sayan3sarkar commented:
> Hi @Sayan3sarkar, thank you for giving the code a beautiful structure! I am getting the following error on trying to run the demo -
> 
> ```
> Not Found
> 404
> NotFoundError: Not Found
>     at /home/xprilion/projects/demoadmin/app.js:46:8
>     at Layer.handle [as handle_request] (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/layer.js:95:5)
>     at trim_prefix (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:317:13)
>     at /home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:284:7
>     at Function.process_params (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:335:12)
>     at next (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:275:10)
>     at /home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:635:15
>     at next (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:260:14)
>     at Function.handle (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:174:3)
>     at router (/home/xprilion/projects/demoadmin/node_modules/express/lib/router/index.js:47:12)
> ```
> 
> Steps to reproduce:
> 
> 1. `npm start`
> 2. Go to `localhost:1111`
> 3. Click on **Login**
> 
> Can you please see why this might be happening?

Hi @xprilion , I have added the missing route, which was causing the app to break. Kindly run a demo, for your convenience. I did not have the credentials to log in, so couldn't test past the login page navigation. Let me know if any other problem arises. I have also updated app.js in 2 places where I have used ES6+ arrow functions in place of the normal 'function' of JavaScript.
